src/main/java/org/fenixedu/academic/domain/accounting/InterestRate.java

```diff
@@ -0,0 +1,224 @@
+package org.fenixedu.academic.domain.accounting;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.fenixedu.bennu.core.domain.Bennu;
+import org.joda.time.Days;
+import org.joda.time.Interval;
+import org.joda.time.LocalDate;
+
+import pt.ist.fenixframework.FenixFramework;
+
+public class InterestRate extends InterestRate_Base {
+
+    private static final BigDecimal NUMBER_OF_DAYS_PER_YEAR = BigDecimal.valueOf(365);
+
+    public InterestRate() {
+        super();
+    }
+
+    private InterestRate(LocalDate start, LocalDate end, BigDecimal value) {
+        super();
+        setRootDomainObject(Bennu.getInstance());
+        setStart(start);
+        setEnd(end);
+        setValue(value);
+    }
+
+    public Interval getInterval() {
+        return new Interval(getStart().toDateTimeAtStartOfDay(), getEnd().toDateTimeAtStartOfDay().plusDays(1));
+    }
+
+    /**
+     * returns whether this interest rate applies to the specified interval
+     * @param interval
+     * @return
+     */
+    public boolean isActive(Interval interval) {
+        return getNumberOfDays(interval) != 0;
+    }
+
+    /**
+     * Returns the number of days that the interval overlaps in this interest rate, 0 if it does not overlaps.
+     * @param interval
+     * @return
+     */
+    public Integer getNumberOfDays(Interval interval) {
+        Interval overlap = getInterval().overlap(interval);
+        return overlap == null ? 0 : Days.daysBetween(overlap.getStart(), overlap.getEnd()).getDays();
```

Comment:
DateTimes will not always work... because of daylight savings time:
   https://stackoverflow.com/questions/3802893/number-of-days-between-two-dates-in-joda-time
doing a toLocaleDate should solve this issue.